python-modules/chevah-python-test/test_python_binary_dist.py

```diff
@@ -3,99 +3,278 @@
+    # libssl.so when checking for OpenSSL.
+    unwanted_deps = []
+    for single_actual_dep in actual_deps:
+        for single_expected_dep in expected_deps:
+            if single_expected_dep in single_actual_dep:
+                break
+        else:
+            unwanted_deps.append(single_actual_dep)
+    return unwanted_deps
+
+def main():
+    """
+    Launches tests to check required modules and OS-specific dependencies.
+    Exits with a relevant error code.
+    """
+    exit_code = 0
+
+    try:
+        import zlib
+        zlib
+    except:
+        print '"zlib" missing.'
+        exit_code = 1
+
+    try:
+        import _hashlib
+        _hashlib
+    except:
+        print 'standard "ssl" missing.'
+        exit_code = 1
+
+    try:
+        from OpenSSL import SSL, crypto, rand
+        SSL
+        crypto
+        rand
+    except:
+        print '"OpenSSL" missing.'
+        exit_code = 1
+
+    try:
+        import Crypto
+        Crypto
+    except:
+        print '"PyCrypto" missing.'
+        exit_code = 1
+
+
+    try:
+        import crypt
+        crypt
+    except:
+        print '"crypt" missing.'
+        exit_code = 1
+
+    try:
+        import pysqlite2
+        pysqlite2
+    except:
+        print '"pysqlite2" missing.'
+        exit_code = 1
+
     try:
-        from ctypes import windll
-        windll
+        import setproctitle
+        setproctitle
     except:
-        print '"ctypes - windll" missing.'
+        print '"setproctitle" missing.'
         exit_code = 1
 
-# Linux specific modules.
-if platform_system == 'linux':
-    # On Linux we need spwd... but not on all Unix system, ex: AIX
     try:
-        import spwd
-        spwd
+        from ctypes import CDLL
+        CDLL
     except:
-        print 'spwd missing.'
+        print '"ctypes - CDLL" missing.'
         exit_code = 1
 
-sys.exit(exit_code)
+    try:
+        from ctypes.util import find_library
+        find_library
+    except:
+        print '"ctypes.utils - find_library" missing.'
+        exit_code = 1
+
+    try:
+        from Crypto.PublicKey import _fastmath
+        _fastmath
+    except:
+        print 'Crypto.PublicKey._fastmath missing. No GMP?'
+        exit_code = 1
+
+    # Windows specific modules.
+    if os.name == 'nt':
+        try:
+            from ctypes import windll
+            windll
+        except:
+            print '"ctypes - windll" missing.'
+            exit_code = 1
+
+    if ( platform_system == 'linux' ) or ( platform_system == 'sunos' ):
+        # On Linux and Solaris we need spwd, but not on AIX or OS X.
+        try:
+            import spwd
+            spwd
+        except:
+            print 'spwd missing.'
+            exit_code = 1
+
+    # Finally, compare the list of expected deps for the current OS with the
+    # list of actual deps returned by the shell script helper.
+    expected_deps = set_expected_deps()
```

Comment:
set_expected_deps or get_expected_deps ? or get_allowed_deps or something like this